### ISSUE TITLE
RangePicker if end date is before start date, should set start date to end date and keep focus in end date

### ISSUE THREAD

**Issue by kevinguebert** [ant-design/ant-design#36690]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-reproduction-template-forked-lzjk2j?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce

1. Add a start date (07/20/2022)
2. Add an end date (07/22/2022)
3. Edit the *end date* to be *before the start date* (end date -> 07/01/2022)
4. Notice the start date is set to the end date but still requires another click to set the date. (In our `onCalendarChange` we set the start date to be the end date if the end date is before the start date)
5. However the start date isn't *actually* set and the cursor remains in the start date

### What is expected?

If the date values are set programmatically, the values get set and the cursors input is based on the values. If we programmatically set the start date, the cursor would be in the end input. 

### What is actually happening?

The cursor does not change locations based on values set. When the date values are set, an extra click is needed to actually set the start value. 

| Environment | Info |
| --- | --- |
| antd | 4.21.7 |
| React | 17.02 |
| System | Mac |
| Browser | Chrome |

---
We are trying to reduce the number of clicks a user needs to take for editing in new dates. If a user changes any of the dates, we want the range picker to be responsive enough based on what dates they choose. A similar user experience as other range picker libraries such as: [https://react-dates.github.io/react-dates/?path=/story/daterangepicker-drp--default](https://react-dates.github.io/react-dates/?path=/story/daterangepicker-drp--default)

**Comment by pookee** [comment]
Just wanted to say we have the exact same issue, it's confusing our users a lot, we had to reverse to 2 separate components for now to avoid confusion.

It would be greatly appreciated to have an option to stop the automatic switch between start and end date :)

**Comment by kevinguebert** [comment]
@pookee This may not be the exact solution you are looking for but I think we've managed to fork the `rc-picker` component and add the functionality. 

We updated the `triggerChange` functionality to check which date was changed and when to switch input focus here: https://github.com/react-component/picker/blob/master/src/RangePicker.tsx#L436-L442

**Comment by pookee** [comment]
@kevinguebert thanks a lot we'll look into it !

**Comment by adventofryz** [comment]
_"if end date is before start date, should set start date to end date"._

This is really bad user experience. You shouldn't automatically set dates without at least telling the user that something has changed. That's why validation errors on select exist.

Currently the behavior in Rangepicker (5.3.2) is as follows:
If you select two dates, with the same day and set start time to 3:00pm and end time to 1:00 pm, the times get reversed automatically without any user intervention. 

I would suggest adding the "order" prop from TimePicker component to Rangepicker to prevent swapping of times and keep automatic ordering to false by default.

**Comment by kevinguebert** [comment]
Good points. I will admit we don't use the time portion of the range picker right now so I did not fully consider how the interaction would work there.

Our inspiration for the ticket came from the UX from Airbnb.com. If a user is selecting a booking date and the end date is before the start date, it updates the start date and keeps the focus in the end date. After watching our users use the range picker, we learned that they expected the Airbnb-like experience.
